version: '3'

# volumes:
#   dbdata:
#     driver: local
#     driver_opts:
#       type: 'none'
#       o: 'bind'
#       device: '/srv/db-data'

services:
  portainer:
    container_name: "portainer"
    image: "portainer/portainer-ce:alpine"
    security_opt:
      - no-new-privileges:true
    command: --no-analytics
    restart: unless-stopped
    networks:
      - proxy
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ${DOCKER_ROOT}/portainer/data:/data
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.portainer.entrypoints=http"
      - "traefik.http.routers.portainer.rule=Host(`${SUBDOMAIN}.${DOMAIN}`)"
      - "traefik.http.routers.portainer.middlewares=secured@file"
      - "traefik.http.routers.portainer-secure.entrypoints=https"
      - "traefik.http.routers.portainer-secure.rule=Host(`${SUBDOMAIN}.${DOMAIN}`)"
      - "traefik.http.routers.portainer-secure.tls=true"
      - "traefik.http.routers.portainer-secure.tls.certresolver=le-dns"
      - "traefik.http.routers.portainer-secure.service=portainer"
      - "traefik.http.services.portainer.loadbalancer.server.port=9000"

networks:
  proxy:
    external: true